src/coreclr/pal/inc/pal.h

```diff
@@ -3642,13 +3642,34 @@ Define_InterlockMethod(
     __atomic_exchange_n(Target, Value, __ATOMIC_ACQ_REL)
 )
 
+#if defined(HOST_X86)
+
+// 64-bit __atomic_exchange_n is not expanded as a compiler intrinsic on Linux x86.
+// Use inline implementation instead.
```

Comment:
Should this maybe do a CMake test for `__atomic_exchange_n` instead?

Reply:
CoreCLR build system not setup to propagate results CMake tests outside the CoreCLR PAL. Getting this wired up would be quite a project.

I hope that we will have get rid of the CoreCLR PAL eventually and we will be able to use the standard C/C++ atomics instead.